api/base/utils.py

```diff
@@ -1,22 +1,24 @@
-import urlparse
+# -*- coding: utf-8 -*-
+
 import furl
-from modularodm.exceptions import NoResultsFound
-from modularodm import Q
+
 from rest_framework.exceptions import NotFound
 from rest_framework.reverse import reverse
-from django.utils.http import urlencode
+from modularodm.exceptions import NoResultsFound
+from modularodm import Q
+
+from api.base import settings as api_settings
+from website import settings as website_settings
+from website import util as website_util  # noqa
 
-from website import settings
 
 def absolute_reverse(view_name, query_kwargs=None, args=None, kwargs=None):
     """Like django's `reverse`, except returns an absolute URL. Also add query parameters."""
     relative_url = reverse(view_name, kwargs=kwargs)
 
-    if query_kwargs:
-        relative_url = '{}?{}'.format(relative_url, urlencode(query_kwargs))
-
-    domain = settings.API_DOMAIN
-    return urlparse.urljoin(domain, relative_url)
+    url = website_util.api_v2_url(relative_url, params=query_kwargs,
```

Comment:
I'm a little surprised this works on staging2. On a page such as https://staging2.osf.io/api/v2/nodes/ , I'd expect that joining the django URL to api domain would yield something like:

`https://staging2.osf.io/v2/nodes/23znr/` (note the missing "api/" in the middle)

In fact, the links.self field for node detail presents the correct URL, such as:  `https://staging2.osf.io/api/v2/nodes/23znr/`

Possibly some custom config, but not sure.